diff --git a/Bloom.sln b/Bloom.sln
index c920c20d3..fd31af9df 100644
--- a/Bloom.sln
+++ b/Bloom.sln
@@ -5,8 +5,6 @@ VisualStudioVersion = 15.0.26430.14
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BloomExe", "src\BloomExe\BloomExe.csproj", "{304D5612-167C-4725-AF27-B9F2BB788B57}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BloomTests", "src\BloomTests\BloomTests.csproj", "{57E7AC72-D901-4A80-A50B-71319AC96CDC}"
-EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution", "Solution", "{04AF3A87-C943-4731-BE3D-659EA705171D}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "DistFiles", "DistFiles", "{D38FC737-8C27-46CB-B490-3CB2E7AB3C99}"
@@ -30,9 +28,11 @@ Global
 		Debug|Any CPU = Debug|Any CPU
 		Debug|Mixed Platforms = Debug|Mixed Platforms
 		Debug|x86 = Debug|x86
+		Debug|x64 = Debug|x64
 		Release|Any CPU = Release|Any CPU
 		Release|Mixed Platforms = Release|Mixed Platforms
 		Release|x86 = Release|x86
+		Release|x64 = Release|x64
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Debug|Any CPU.ActiveCfg = Debug|x86
@@ -41,24 +41,16 @@ Global
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Debug|Mixed Platforms.Build.0 = Debug|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Debug|x86.ActiveCfg = Debug|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Debug|x86.Build.0 = Debug|x86
+		{304D5612-167C-4725-AF27-B9F2BB788B57}.Debug|x64.ActiveCfg = Debug|x64
+		{304D5612-167C-4725-AF27-B9F2BB788B57}.Debug|x64.Build.0 = Debug|x64
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|Any CPU.ActiveCfg = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|Any CPU.Build.0 = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|Mixed Platforms.ActiveCfg = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|Mixed Platforms.Build.0 = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|x86.ActiveCfg = Release|x86
 		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|x86.Build.0 = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Any CPU.ActiveCfg = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Any CPU.Build.0 = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|Mixed Platforms.Build.0 = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|x86.ActiveCfg = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Debug|x86.Build.0 = Debug|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Any CPU.ActiveCfg = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Any CPU.Build.0 = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Mixed Platforms.ActiveCfg = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|Mixed Platforms.Build.0 = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|x86.ActiveCfg = Release|x86
-		{57E7AC72-D901-4A80-A50B-71319AC96CDC}.Release|x86.Build.0 = Release|x86
+		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|x64.ActiveCfg = Release|x64
+		{304D5612-167C-4725-AF27-B9F2BB788B57}.Release|x64.Build.0 = Release|x64
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
diff --git a/debian/bloom-beta b/debian/bloom-beta
index 3e6be8764..8af398548 100755
--- a/debian/bloom-beta
+++ b/debian/bloom-beta
@@ -1,11 +1,21 @@
 #!/bin/sh
 
-LIB=/usr/lib/bloom-desktop-beta
-SHARE=/usr/share/bloom-desktop-beta
+LIB=/app/lib/bloom-desktop
+SHARE=/app/share/bloom-desktop
 
 cd "$SHARE"
 RUNMODE=INSTALLED
 . ./environ
 cd "$OLDPWD"
 
-exec "$LIB"/BloomLauncher "$@"
+# needed for Sldr.Initialize() in flatpak
+mkdir -p /var/lock
+
+# maximize exposure of user fonts to Bloom
+[ -n "$XDG_DATA_HOME" ] && mkdir -p "$XDG_DATA_HOME/fonts"
+[ -n "$XDG_DATA_HOME" -a ! -e "$XDG_DATA_HOME/fonts/local" ] && ln -s "$HOME/.local/share/fonts" "$XDG_DATA_HOME/fonts/local"
+[ -n "$XDG_DATA_HOME" -a ! -e "$XDG_DATA_HOME/fonts/app" ] && ln -s "/app/share/fonts" "$XDG_DATA_HOME/fonts/app"
+
+# Without this sync, communication with BloomLibrary (or at least S3) will fail.
+[ ! -d ~/.var/app/org.sil.Bloom/config/.mono/certs ] && /app/bin/mono /app/lib/mono/4.5/cert-sync.exe --user /app/share/ca-certificates/ca-certificates.crt
+exec /app/bin/mono --debug "$LIB"/Bloom.exe "$@"
diff --git a/debian/bloom-beta.appdata.xml b/debian/bloom-beta.appdata.xml
index e6fa74dc0..de0b28b16 100644
--- a/debian/bloom-beta.appdata.xml
+++ b/debian/bloom-beta.appdata.xml
@@ -38,7 +38,48 @@
   <url type="bugtracker">https://issues.bloomlibrary.org/youtrack</url>
   <url type="help">https://community.software.sil.org/c/bloom</url>
   <releases>
-    <release version="5.4" date="2023-01-10">
+    <release version="5.4.1" date="2023-05-18">
+      <description>
+        <ul>
+          <li>Simple Choice Activities: We’ve added two new activities that you can add to your books. Just enter the correct answer in the first box. When someone reads the book, Bloom will shuffle the answers.  These are Choose Word from Picture and Choose Picture from Word</li>
+          <li>Smarter Splitter: We heard you: getting image and text boxes to just the right size was too hard. We’ve made the splitter smart in a couple of ways: 1) You can now double-click the splitter to make it snap to the position used by the previous page. 2) The splitter now “snaps” to some common locations such as 1/4, 1/3, 1/2, etc.</li>
+          <li>Fixed Layout ePUBs: Bloom can now create ePUBs in “Fixed” mode which asks ePUB players to preserve the page as you see it in Bloom. If you have a very simple book, Bloom will also let you choose the previous “Flowable” mode. With this improvement, you can now publish overlay (comic) books as ePUBs.</li>
+          <li>Color Chooser: We improved the color chooser and made the same chooser available everywhere you can set a color, including text, background colors in the overlay tool, and cover colors.</li>
+          <li>Text Color in Styles:  You can now set color as part of a style definition.  As before, you can also color individual words and letters.</li>
+          <li>“Needs Copyright” Button: We’ve noticed that sometimes people forget to add a copyright, which later prevents them from publishing the book to BloomLibrary.org. Now Bloom shows a button on the Credits page to prompt you.</li>
+          <li>Overlay Tool: We’ve added a new Bubble shape, “Rectangular”. You can now click on the background image in order to change it. If a bubble somehow ends up off-screen, Bloom will move it back on screen the next time it loads the page. (Note that a Bloom Enterprise subscription is required to add new Overlay (comic) images to a book.)</li>
+        </ul>
+      </description>
+    </release>
+    <release version="5.3.3" date="2023-05-11">
+      <description>bug fixes</description>
+    </release>
+    <release version="5.3.2" date="2023-02-10">
+      <description>bug fixes</description>
+    </release>
+    <release version="5.3.1" date="2023-01-11">
+      <description>
+        <ul>
+          <li>Publish to Audio or Video: You can now turn your book into a video that can be shared on Facebook or YouTube. You can also make a version that will work on inexpensive “feature phones” or mp3 players.</li>
+          <li>Text Color: To highlight a word or some characters, you can now select some text and set its color.</li>
+          <li>New Copyright / License Dialog: We’ve re-worked the copyright / license dialog in order to help you better understand your choices.</li>
+          <li>Font License Info: When you’re just making paper or PDF copies of your book, you can use whatever fonts are on your computer. However, many fonts are not freely licensed for use on the web or in ebooks. But how are you supposed to know? It turns out that inside many fonts, there is some metadata that indicates things like what the license is, where the font comes from, and what version of the font you have. Bloom now reads this information and uses color and icons to indicate how safe a font is to use.</li>
+        </ul>
+      </description>
+    </release>
+    <release version="5.2.2" date="2022-11-29">
+      <description>bug fixes</description>
+    </release>
+    <release version="5.2.1" date="2022-07-07">
+      <description>
+        <ul>
+          <li>Decodable Reader: When the Decodable Reader tool generates a report, it now includes a list of all the usable words for each level.</li> 
+          <li>Uploading to bloomlibrary.org: When you upload a book, Bloom checks to see if there is already a book with the same internal ID on the library. If it finds one, it now gives you two choices. You can replace the existing book or upload your book as a new item. The latter option fixes your book's ID to be unique.</li>
+          <li>Team Collections (Bloom Enterprise, Experimental): User can now enter checkin notes and view a history of checkin notes.  Administrators can forceably unlock a book that has been left checked out.</li>
+          <li>Export / Import Spreadsheet (Bloom Enterprise, Experimental): Bloom books can now be exported to / imported from Excel spreadsheets. Some users will find this helpful for large translation projects or for making bulk changes.</li>
+          <li>Collection Settings: Many Bloom Enterprise projects use their own custom front/back matter packs. Previously in Bloom, everyone could see these in the list of choices. Now, these custom choices are not listed but instead are automatically selected based on the Bloom Enterprise code.</li>
+        </ul>
+      </description>
     </release>
   </releases>
   <content_rating type="oars-1.0"/>
diff --git a/environ b/environ
index 44b1f73b2..1673c7b5f 100644
--- a/environ
+++ b/environ
@@ -6,9 +6,9 @@
 # Note that this file is intended to be "sourced", not "executed".
 
 # the sourcing script should cd/pushd to the directory containing this script
-BASE="$(pwd)"
+BASE="$(pwd | sed s=/app/share/=/app/lib/=)"
 [ -z "$BUILD" ] && BUILD=Debug
-[ -z "$MONO_PREFIX" ] && MONO_PREFIX=/opt/mono5-sil
+[ -z "$MONO_PREFIX" ] && MONO_PREFIX=/app
 
 # Dependency locations
 # Search for xulrunner and geckofx, select the best, and add its location to LD_LIBRARY_PATH.
@@ -26,18 +26,18 @@ MONO_ENV_OPTIONS="-O=-gshared"
 # chmsee was phased out by Ubuntu in bionic (18.04), and supporting libraries by disco (19.04)
 
 if [ -z "$MONO_HELP_VIEWER" ]; then
-	if [ -f /usr/bin/chmsee ]; then
+	if [ -f /app/bin/chmsee ]; then
 		export MONO_HELP_VIEWER=chmsee
-	elif [ -f /usr/bin/xchm ]; then
+	elif [ -f /app/bin/xchm ]; then
 		export MONO_HELP_VIEWER=xchm
-	elif [ -f /usr/bin/kchmviewer ]; then
+	elif [ -f /app/bin/kchmviewer ]; then
 		export MONO_HELP_VIEWER=kchmviewer
 	fi
 fi
 
 ################################################################################################
 
-if [ "$RUNMODE" = "INSTALLED" ]
+if [ "$RUNMODE" = "INSTALLED" -o "$FLATPAK_ID" = "org.sil.Bloom" ]
 then
 	# Add the installation directory to the library paths
 	LD_LIBRARY_PATH="${BASE}:${LD_LIBRARY_PATH}"
@@ -55,7 +55,7 @@ fi
 
 ################################################################################################
 
-if [ "$RUNMODE" = "PACKAGE" -o "$RUNMODE" = "INSTALLED" ]
+if [ "$RUNMODE" = "PACKAGE" -o "$RUNMODE" = "INSTALLED" -o "$FLATPAK_ID" = "org.sil.Bloom" ]
 then
 	# Add packaged mono items to paths
 	PATH="${MONO_PREFIX}/bin:${PATH}"
@@ -73,9 +73,6 @@ else
 	MONO_GAC_PREFIX="${MONO_PREFIX}:/usr"
 fi
 
-# point to our private version of libtidy
-LD_LIBRARY_PATH="/opt/tidy-sil/lib:${LD_LIBRARY_PATH}"
-
 ################################################################################################
 
 MONO_TRACE_LISTENER="Console.Out"
@@ -101,3 +98,7 @@ export HGRCPATH=
 
 #sets keyboard input method to none
 unset XMODIFIERS
+
+if [ "$DEBUG_BLOOM" = "true" ]; then
+	env | sort
+fi
diff --git a/environ-xulrunner b/environ-xulrunner
index 91b1fa261..808a03c23 100644
--- a/environ-xulrunner
+++ b/environ-xulrunner
@@ -2,8 +2,8 @@
 # for them.  Note that this script lives in /usr/share/bloom-desktop* on user machines,
 # but on the root of the repository on developer machines.  This requires a bit of
 # hackery to reliably find the proper location of the Firefox/xulrunner libraries.
-tmpbasedir=$(pwd | sed s=usr/share/=usr/lib/=)
-if (echo $tmpbasedir | grep /usr/lib/bloom-desktop >/dev/null); then
+tmpbasedir=$(pwd | sed s=app/share/=app/lib/=)
+if (echo $tmpbasedir | grep /app/lib/bloom-desktop >/dev/null); then
   export XULRUNNER=$(dirname $(find $tmpbasedir -name libxul.so))
 else
   export XULRUNNER=$(dirname $(find ${tmpbasedir}/output/${BUILD} -name libxul.so))
diff --git a/src/BloomBrowserUI/gulpfile.js b/src/BloomBrowserUI/gulpfile.js
index 5084f87ed..b23ce38cf 100644
--- a/src/BloomBrowserUI/gulpfile.js
+++ b/src/BloomBrowserUI/gulpfile.js
@@ -102,7 +102,9 @@ var paths = {
 var allXliffFiles = globule.find(paths.xliff);
 // Check for the existence of the SIL mono, which flags we're on Linux
 // and need to use it to execute HtmlXliff.exe
-var IsLinux = globule.find(["/opt/mono5-sil/**/mono"]).length > 0;
+var IsLinuxWithMono5 = globule.find(["/opt/mono5-sil/**/mono"]).length > 0;
+// Check for files in /etc which flag we're on Linux, possibly with the system mono.
+var IsLinux = globule.find(["/etc/os-release","/etc/passwd","/etc/group"]).length === 3;
 
 gulp.task("less", function() {
     var less = require("gulp-less");
@@ -296,9 +298,11 @@ gulp.task("translateHtmlFiles", function() {
                     var xliffFile = xliffFiles[i]; // needed for error message to work
                     var outfile = getOutputFilename(file.path, xliffFile);
                     var cmd = "";
-                    if (IsLinux)
+                    if (IsLinuxWithMono5)
                         cmd =
                             "/opt/mono5-sil/bin/mono --debug ../../lib/dotnet/HtmlXliff.exe --inject";
+		    else if (IsLinux)
+		        cmd = "mono --debug  ../../lib/dotnet/HtmlXliff.exe --inject";
                     else cmd = "..\\..\\lib\\dotnet\\HtmlXliff.exe --inject";
                     cmd = cmd + ' -x "' + xliffFile + '"';
                     cmd = cmd + ' -o "' + outfile + '"';
@@ -324,9 +328,11 @@ gulp.task("createXliffFiles", function() {
             tap(function(file) {
                 var xliffFile = getXliffFilename(file.path);
                 var cmd = "";
-                if (IsLinux)
+                if (IsLinuxWithMono5)
                     cmd =
                         "/opt/mono5-sil/bin/mono --debug ../../lib/dotnet/HtmlXliff.exe --extract --preserve";
+		else if (IsLinux)
+		    cmd = "mono --debug  ../../lib/dotnet/HtmlXliff.exe --extract --preserve";
                 else
                     cmd =
                         "..\\..\\lib\\dotnet\\HtmlXliff.exe --extract --preserve";
diff --git a/src/BloomExe/BloomExe.csproj b/src/BloomExe/BloomExe.csproj
index 2cdc78b54..cbc65e5df 100644
--- a/src/BloomExe/BloomExe.csproj
+++ b/src/BloomExe/BloomExe.csproj
@@ -10,7 +10,7 @@
     <TargetFrameworkProfile>
     </TargetFrameworkProfile>
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
-    <RestorePackages>true</RestorePackages>
+    <RestorePackages>false</RestorePackages>
     <PackagesConfig Condition="'$(OS)'=='Windows_NT'">$(MSBuildProjectDirectory)\packages.config</PackagesConfig>
     <PackagesConfig Condition="'$(OS)'!='Windows_NT'">$(MSBuildProjectDirectory)/Linux/packages.config</PackagesConfig>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -68,6 +68,54 @@
   <PropertyGroup Condition="'$(OS)'!='Windows_NT' AND '$(Configuration)|$(Platform)' == 'Release|x86'">
     <DefineConstants>__MonoCS__;TRACE</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">
+    <DebugSymbols>true</DebugSymbols>
+    <OutputPath>..\..\output\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <DebugType>full</DebugType>
+    <PlatformTarget>x64</PlatformTarget>
+    <CodeAnalysisLogFile>..\..\output\Debug\Bloom.exe.CodeAnalysisLog.xml</CodeAnalysisLogFile>
+    <CodeAnalysisUseTypeNameInSuppression>true</CodeAnalysisUseTypeNameInSuppression>
+    <CodeAnalysisModuleSuppressionsFile>GlobalSuppressions.cs</CodeAnalysisModuleSuppressionsFile>
+    <ErrorReport>prompt</ErrorReport>
+    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSetDirectories>;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Team Tools\Static Analysis Tools\\Rule Sets</CodeAnalysisRuleSetDirectories>
+    <CodeAnalysisIgnoreBuiltInRuleSets>false</CodeAnalysisIgnoreBuiltInRuleSets>
+    <CodeAnalysisRuleDirectories>;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Team Tools\Static Analysis Tools\FxCop\\Rules</CodeAnalysisRuleDirectories>
+    <CodeAnalysisIgnoreBuiltInRules>false</CodeAnalysisIgnoreBuiltInRules>
+    <CodeAnalysisFailOnMissingRules>false</CodeAnalysisFailOnMissingRules>
+    <WarningLevel>4</WarningLevel>
+    <Optimize>false</Optimize>
+    <Prefer32Bit>false</Prefer32Bit>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(OS)'!='Windows_NT' AND '$(Configuration)|$(Platform)' == 'Debug|x64'">
+    <DefineConstants>__MonoCS__;DEBUG;TRACE</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
+    <OutputPath>..\..\output\Release\</OutputPath>
+    <DebugType>pdbonly</DebugType>
+    <PlatformTarget>x64</PlatformTarget>
+    <CodeAnalysisLogFile>..\..\output\Release\Bloom.exe.CodeAnalysisLog.xml</CodeAnalysisLogFile>
+    <CodeAnalysisUseTypeNameInSuppression>true</CodeAnalysisUseTypeNameInSuppression>
+    <CodeAnalysisModuleSuppressionsFile>GlobalSuppressions.cs</CodeAnalysisModuleSuppressionsFile>
+    <ErrorReport>prompt</ErrorReport>
+    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSetDirectories>;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Team Tools\Static Analysis Tools\\Rule Sets</CodeAnalysisRuleSetDirectories>
+    <CodeAnalysisIgnoreBuiltInRuleSets>true</CodeAnalysisIgnoreBuiltInRuleSets>
+    <CodeAnalysisRuleDirectories>;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Team Tools\Static Analysis Tools\FxCop\\Rules</CodeAnalysisRuleDirectories>
+    <CodeAnalysisIgnoreBuiltInRules>false</CodeAnalysisIgnoreBuiltInRules>
+    <CodeAnalysisFailOnMissingRules>false</CodeAnalysisFailOnMissingRules>
+    <WarningLevel>4</WarningLevel>
+    <Optimize>false</Optimize>
+    <DebugSymbols>true</DebugSymbols>
+    <DefineConstants>TRACE</DefineConstants>
+    <Prefer32Bit>false</Prefer32Bit>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(OS)'!='Windows_NT' AND '$(Configuration)|$(Platform)' == 'Release|x64'">
+    <DefineConstants>__MonoCS__;TRACE</DefineConstants>
+  </PropertyGroup>
   <PropertyGroup>
     <OutputType>WinExe</OutputType>
   </PropertyGroup>
@@ -185,7 +233,7 @@
     </Reference>
     <Reference Include="MarkdownDeep, Version=1.5.4615.26275, Culture=neutral, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\packages\MarkdownDeep.NET.1.5\lib\.NetFramework 3.5\MarkdownDeep.dll</HintPath>
+      <HintPath>..\..\packages\MarkdownDeep.NET.1.5.0\lib\.NetFramework 3.5\MarkdownDeep.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Bcl.AsyncInterfaces, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Microsoft.Bcl.AsyncInterfaces.7.0.0\lib\net462\Microsoft.Bcl.AsyncInterfaces.dll</HintPath>
@@ -240,7 +288,7 @@
     <Reference Include="PdfSharp">
       <HintPath>..\..\lib\dotnet\PdfSharp.dll</HintPath>
     </Reference>
-    <Reference Include="PodcastUtilities.PortableDevices">
+    <Reference Include="PodcastUtilities.PortableDevices" Condition="'$(OS)'=='Windows_NT'">
       <HintPath>..\..\lib\dotnet\PodcastUtilities.PortableDevices.dll</HintPath>
     </Reference>
     <Reference Include="PresentationCore" />
@@ -286,7 +334,7 @@
     <Reference Include="Spart, Version=1.0.0.0, Culture=neutral, PublicKeyToken=14d24e064e474331, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Spart.1.0.0\lib\net461\Spart.dll</HintPath>
     </Reference>
-    <Reference Include="Splat">
+    <Reference Include="Splat" Condition="'$(OS)'=='Windows_NT'">
       <HintPath>..\..\lib\dotnet\Splat.dll</HintPath>
     </Reference>
     <Reference Include="SQLite-net, Version=1.7.335.0, Culture=neutral, processorArchitecture=MSIL">
@@ -308,6 +356,9 @@
       <HintPath>..\..\lib\dotnet\Squirrel.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Buffers, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Buffers.4.5.1\lib\net461\System.Buffers.dll</HintPath>
+    </Reference>
     <Reference Include="System.Collections.Immutable, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.Collections.Immutable.5.0.0\lib\net461\System.Collections.Immutable.dll</HintPath>
     </Reference>
@@ -330,6 +381,9 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="System.IO.Compression.FileSystem" />
+    <Reference Include="System.IO.FileSystem, Version=4.0.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.IO.FileSystem.4.3.0\lib\net46\System.IO.FileSystem.dll</HintPath>
+    </Reference>
     <Reference Include="System.IO.FileSystem.AccessControl, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.IO.FileSystem.AccessControl.5.0.0\lib\net461\System.IO.FileSystem.AccessControl.dll</HintPath>
     </Reference>
@@ -358,6 +412,9 @@
     <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.Runtime.CompilerServices.Unsafe.6.0.0\lib\net461\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
     </Reference>
+    <Reference Include="System.Runtime.InteropServices.RuntimeInformation, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Runtime.InteropServices.RuntimeInformation.4.3.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll</HintPath>
+    </Reference>
     <Reference Include="System.Runtime.Serialization" />
     <Reference Include="System.Numerics.Vectors">
       <HintPath>..\..\packages\System.Numerics.Vectors.4.5.0\lib\net46\System.Numerics.Vectors.dll</HintPath>
@@ -382,9 +439,10 @@
     </Reference>
     <Reference Include="System.Threading.Tasks, Version=4.0.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />
     <Reference Include="System.Transactions" />
-    <Reference Include="System.ValueTuple, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\System.ValueTuple.4.5.0\lib\net47\System.ValueTuple.dll</HintPath>
+    <Reference Include="System.Threading.Tasks.Extensions, Version=4.2.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Threading.Tasks.Extensions.4.5.4\lib\net461\System.Threading.Tasks.Extensions.dll</HintPath>
     </Reference>
+    <Reference Include="System.ValueTuple" />
     <Reference Include="System.Web" />
     <Reference Include="System.Web.Extensions" />
     <Reference Include="System.Xml.Linq" />
diff --git a/src/BloomExe/Linux/packages.config b/src/BloomExe/Linux/packages.config
index b0ffef59b..261e7a9f0 100644
--- a/src/BloomExe/Linux/packages.config
+++ b/src/BloomExe/Linux/packages.config
@@ -27,7 +27,7 @@
   <package id="JsonFx" version="2.0.1209.2802" targetFramework="net461" />
   <package id="L10NSharp" version="6.0.0" targetFramework="net472" />
   <package id="Markdig.Signed" version="0.30.4" targetFramework="net472" />
-  <package id="MarkdownDeep.NET" version="1.5" targetFramework="net40-Client" />
+  <package id="MarkdownDeep.NET" version="1.5.0" targetFramework="net40-Client" />
   <package id="Microsoft.Bcl.AsyncInterfaces" version="7.0.0" targetFramework="net472" />
   <package id="Microsoft.CSharp" version="4.7.0" targetFramework="net472" />
   <package id="Microsoft.DotNet.PlatformAbstractions" version="3.1.6" targetFramework="net461" />
@@ -89,4 +89,7 @@
   <!-- Linux specific packages.  Changes above this line must be copied to ../packages.config -->
   <package id="Geckofx60.64.Linux" version="60.0.51" targetFramework="net461" />
   <package id="SharpFont" version="3.1.0" targetFramework="net461" />
+  <!-- copied from BloomTests -->
+  <package id="System.IO.FileSystem" version="4.3.0" targetFramework="net461" />
+  <package id="System.Threading.Tasks.Extensions" version="4.5.4" targetFramework="net461" />
 </packages>
diff --git a/src/content/package.json b/src/content/package.json
index c8f9a169a..fbd090805 100644
--- a/src/content/package.json
+++ b/src/content/package.json
@@ -48,7 +48,7 @@
     },
     "license": "MIT",
     "dependencies": {
-        "@types/node": "^18.6.3",
+        "@types/node": "16.18.0",
         "typescript": "^4.6.3"
     },
     "packageManager": "yarn@1.22.19",
diff --git a/src/content/yarn.lock b/src/content/yarn.lock
index 1172b6085..8b59fdcec 100644
--- a/src/content/yarn.lock
+++ b/src/content/yarn.lock
@@ -87,10 +87,10 @@
   dependencies:
     "@types/babel-types" "*"
 
-"@types/node@^18.6.3":
-  version "18.6.3"
-  resolved "https://registry.yarnpkg.com/@types/node/-/node-18.6.3.tgz#4e4a95b6fe44014563ceb514b2598b3e623d1c98"
-  integrity sha512-6qKpDtoaYLM+5+AFChLhHermMQxc3TOEFIDzrZLPRGHPrLEwqFkkT5Kx3ju05g6X7uDPazz3jHbKPX0KzCjntg==
+"@types/node@16.18.0":
+  version "16.18.0"
+  resolved "https://registry.npmjs.org/@types/node/-/node-16.18.0.tgz#34982c6d5b5734c900f4ee69e1b87d212db6c60f"
+  integrity sha512-LqYqYzYvnbCaQfLAwRt0zboqnsViwhZm+vjaMSqcfN36vulAg7Pt0T83q4WZO2YOBw3XdyHi8cQ88H22zmULOA==
 
 acorn-globals@^3.0.0:
   version "3.1.0"
